dali/operators/reader/webdataset_reader_op.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/webdataset_reader_op.h"
+#include <algorithm>
+#include <string>
+
+namespace dali {
+
+void WebdatasetReader::RunImpl(HostWorkspace& ws) {
+  int num_outputs = ws.NumOutput();
+  int num_samples = GetCurrBatchSize();
+
+  for (int data_idx = 0; data_idx < num_samples; data_idx++) {
+    auto& sample = GetSample(data_idx);
+    for (int output_idx = 0; output_idx < num_outputs; output_idx++) {
+      ws.OutputRef<CPUBackend>(output_idx)[data_idx].ShareData(&sample[output_idx]);
```

Comment:
I'm not sure if it is safe.
`ReadOne` from the loader.h returns `LoadTargetSharedPtr`, which is stored in `prefetched_batch_queue_`.
`GetSample` returns a `*LoadTargetSharedPtr`.
When you continue reading `prefetched_batch_queue_` is trashed, `LoadTargetSharedPtr` calls its custom deleter which moves underlying tensor to `empty_tensors_` ([see](https://github.com/NVIDIA/DALI/blob/main/dali/operators/reader/loader/loader.h#L176)).
So sharing memory with something what can be trashed doesn't sound like a safe thing to do.

Reply:
I think other readers do parsing and do not directly share the tensors they get from `GetSample` (at least I cannot recall nay example now). But I can be wrong...